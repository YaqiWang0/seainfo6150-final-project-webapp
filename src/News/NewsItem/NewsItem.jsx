import React from "react"
import style from "./NewsItem.module.css"

const NewsItem = (props) => {

    const click = () => {
        window.open(props.data.url, "_blank")
    }
    return (
        <div className={style.container} onClick={click}>
            <img  src={props.data.image} alt={props.data.title} className={style.pic}/>
            <div className={style.contentBlock}>
                <div className={style.titleBlock}>
                    <div className={style.title}>{props.data.title}</div>
                    <div className={style.symbol}>{props.data.symbol}</div>
                </div>
                <div className={style.content}>{props.data.text}</div>
            </div>
        </div>
    )
}

export default NewsItem