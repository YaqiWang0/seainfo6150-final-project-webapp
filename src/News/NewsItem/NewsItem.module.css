
.title {
    font-size: 2vmin;
    text-align: left;
    font-weight: bold;
    font-family: "Charter";
    color: white;
}

.symbol {
    font-size: 1.5vmin;
    color: #767676;
    font-family: "Arial Rounded MT Bold";
}

.content {
    font-size: 1em;
    text-align: left;
    text-space: 3vmin;
    color: white;
}

.titleBlock {
    display: flex;
    margin-bottom: 3vmin;
    width: 100%;
}

.contentBlock {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    width: 70%;
    padding-left: 1vmin;
}

@media screen and (min-width: 0px) {
    .pic {
        width: 90%;
        height: 100%;
        max-height: 400px;
    }

    .title {
        margin-bottom: 1vmin;
        width: 100%;
    }

    .container {
        display: flex;
        flex-direction: column;
        border-radius: 5%;
        padding: 1vmin;
        cursor: pointer;
        align-items: center;
    }

    .contentBlock {
        width: 90%;
    }

    .content {
        display: none;
    }

    .titleBlock {
        flex-direction: column;
        justify-content: space-between;
        align-items: flex-start;
    }
}

@media screen and (min-width: 800px) {
    .pic {
        width: 90%;
        height: 100%;
        max-height: 400px;

    }

    .content {
        display: block;
    }

    .container {
        display: flex;
        flex-direction: column;
        border-radius: 5%;
        padding: 1vmin;
        cursor: pointer;
        align-items: center;
    }

    .contentBlock {
        width: 90%;
    }

    .title {
        width: 75%;
        margin-bottom: 0;
    }

    .titleBlock {
        flex-direction: row;
        justify-content: space-between;
        align-items: flex-end;
    }
}

@media screen and (min-width: 1500px) {
    .pic {
        width: 30%;
        height: 100%;
        max-height: 400px;

    }
    .container {
        display: flex;
        flex-direction: row;
        border-radius: 5%;
        padding: 1vmin;
        cursor: pointer;
        align-items: center;
    }

    .contentBlock {
        width: 70%;
    }

    .content {
        display: block;
    }


}