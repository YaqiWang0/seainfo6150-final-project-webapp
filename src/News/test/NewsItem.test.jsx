import React from "react";
import data from "./singleData";
import NewsItem from "../NewsItem/NewsItem";

describe("NewsItem test", () => {

    it('render successfully', function () {
        const {container} = render(
            <NewsItem data = {data}/>
        );
        expect(container).toMatchSnapshot();
    })
})