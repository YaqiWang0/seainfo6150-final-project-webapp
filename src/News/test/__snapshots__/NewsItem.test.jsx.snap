// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsItem test render successfully 1`] = `
<div>
  <div
    class="container"
  >
    <img
      alt="5 Stocks Growing Earnings"
      class="pic"
      src="https://cdn.snapi.dev/images/v1/h/h/bussines-services9.jpg"
    />
    <div
      class="contentBlock"
    >
      <div
        class="titleBlock"
      >
        <div
          class="title"
        >
          5 Stocks Growing Earnings
        </div>
        <div
          class="symbol"
        >
          NSIT
        </div>
      </div>
      <div
        class="content"
      >
        According to the GuruFocus All-in-One Screener, a Premium feature, as of Dec. 11, the following guru-held companies have positive future earnings estimates from Morningstar analysts.
      </div>
    </div>
  </div>
</div>
`;
